src/routes/inviteRouter.ts

```diff
@@ -23,20 +23,16 @@ export class InviteRouter implements RouterInterface {
 
 		router.post(
 			'/batchSend',
-			this._requestAuth.checkLoggedIn,
-			this._requestAuth.checkIsOrganiser,
-			this._inviteController.batchSend.bind(this._inviteController)
+			this._requestAuth.withAuthMiddleware(this, this._inviteController.batchSend)
 		);
 
 		router.put(
 			'/:id([a-f0-9-]+)/send',
-			this._requestAuth.checkLoggedIn,
-			this._requestAuth.checkIsOrganiser,
-			this._inviteController.send.bind(this._inviteController)
+			this._requestAuth.withAuthMiddleware(this, this._inviteController.send)
 		);
 
 		router.get('/:id([a-f0-9-]+)/confirm',
-			this._inviteController.confirm.bind(this._inviteController));
+			this._inviteController.confirm);
```

Comment:
Is authorization not required here?

Reply:
iirc in the email we send just has a link like `https://guh.com/invite/4d084fed-8fa4-4bdb-9215-92d1516bbce0/confirm` which confirms their attendance to the hackathon. When I created it originally I must have thought we didn't need any auth on the route, but maybe we do?

Reply:
It possible (but unlikely) that a user could confirm the attendance of another user by just replacing the ID, right?

Reply:
Yeah, it's technically possible to confirm attendance for someone else, I'll add the auth middleware here just to be safe

Reply:
Right, we're not really going to solve this issue by just adding the auth middleware to this route. Currently we don't have support for resource arguments in the URIs so if we want to give a user access to the confirm operation, we'd have to give them access to using the operation with any arguments, meaning that they'd still be able to just replace the id in the URL.

Let's leave this for after the launch (with high priority) as we're not going to be sending out invitations immediately anyway.